drivers/input/input_ili2132a.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2024 Joel Jaldemark
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ilitek_ili2132a
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/input/input.h>
+#include <zephyr/pm/device.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ili2132a, CONFIG_INPUT_LOG_LEVEL);
+
+#define IS_TOUCHED_BIT 0x40
+
+struct ili2132a_data {
+	const struct device *dev;
+	struct gpio_callback gpio_cb;
+	struct k_work work;
+};
+
+struct ili2132a_config {
+	struct i2c_dt_spec bus;
+	struct gpio_dt_spec rst;
+	struct gpio_dt_spec irq;
+};
+
+static void gpio_isr(const struct device *dev, struct gpio_callback *cb, uint32_t pin)
+{
+	struct ili2132a_data *data = CONTAINER_OF(cb, struct ili2132a_data, gpio_cb);
+
+	k_work_submit(&data->work);
+}
+
+static void ili2132a_process(const struct device *dev)
+{
+	const struct ili2132a_config *dev_cfg = dev->config;
+	uint8_t buf[8];
+	uint16_t x, y;
+
+	i2c_read_dt(&dev_cfg->bus, buf, sizeof(buf));
+	if (buf[1] & IS_TOUCHED_BIT) {
+		x = (buf[3] << 8) + buf[2];
+		y = (buf[5] << 8) + buf[4];
+		input_report_abs(dev, INPUT_ABS_X, x, false, K_FOREVER);
+		input_report_abs(dev, INPUT_ABS_Y, y, false, K_FOREVER);
+		input_report_key(dev, INPUT_BTN_TOUCH, 1, true, K_FOREVER);
+	} else {
+		input_report_key(dev, INPUT_BTN_TOUCH, 0, true, K_FOREVER);
+	}
+}
+
+static void ili2132a_work_handler(struct k_work *work_item)
+{
+	struct ili2132a_data *data = CONTAINER_OF(work_item, struct ili2132a_data, work);
+
+	ili2132a_process(data->dev);
+}
+
+int ili2132a_init(const struct device *dev)
+{
+	struct ili2132a_data *data = dev->data;
+	const struct ili2132a_config *dev_cfg = dev->config;
+	int ret;
+
+	if (!i2c_is_ready_dt(&dev_cfg->bus)) {
+		LOG_ERR("I2C controller device not ready");
+	}
+
+	data->dev = dev;
+
+	ret = gpio_pin_configure_dt(&dev_cfg->irq, GPIO_INPUT);
+	if (ret < 0) {
+		LOG_ERR("Could not configure interrupt gpio");
+		return ret;
+	}
+
+	ret = gpio_pin_configure_dt(&dev_cfg->rst, GPIO_OUTPUT_ACTIVE);
+	if (ret < 0) {
+		LOG_ERR("Could not configure reset gpio");
+		return ret;
+	}
+
+	ret = gpio_pin_set_dt(&dev_cfg->rst, 0);
+	if (ret < 0) {
+		return ret;
+	}
+
+	gpio_init_callback(&data->gpio_cb, gpio_isr, BIT(dev_cfg->irq.pin));
+	ret = gpio_add_callback(dev_cfg->irq.port, &data->gpio_cb);
+	if (ret < 0) {
+		LOG_ERR("Could not set gpio callback");
+		return ret;
+	}
+
+	ret = gpio_pin_interrupt_configure_dt(&dev_cfg->irq, GPIO_INT_EDGE_FALLING);
+	if (ret < 0) {
+		LOG_ERR("Could not configure intterupt");
```

Comment:
typo, `interrupt` (wow, must be the first time I catch a typo rather than make one, I'll give myself a high five)

Reply:
Nice catch, Im always better at finding other peoples typos than my own..